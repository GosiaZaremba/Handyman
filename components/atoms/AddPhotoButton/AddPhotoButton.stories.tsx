import React from 'react';
import {storiesOf} from '@storybook/react-native';
import {AddPhotoButton} from './AddPhotoButton';

storiesOf('atoms/AddPhotoButton', module)
  .addDecorator(story => story())
  .add('default', () => (
    <AddPhotoButton
      onPress={() => {
        console.log('button atom');
      }}
    />
  ));
