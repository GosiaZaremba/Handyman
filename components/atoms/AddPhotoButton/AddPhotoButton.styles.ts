import {StyleSheet} from 'react-native';
import {Colors} from '../../../constants/colors';

export default StyleSheet.create({
  pressed: {
    opacity: 0.9,
    backgroundColor: Colors.font.white,
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',
    top: -30,
    left: 70,
    position: 'relative',
    width: 30,
    height: 30,
  },
  pressable: {
    backgroundColor: Colors.font.white,
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',
    top: -30,
    left: 70,
    position: 'relative',
    width: 30,
    height: 30,
  },
  buttonIcon: {color: Colors.button.addPhoto, fontSize: 26},
});
