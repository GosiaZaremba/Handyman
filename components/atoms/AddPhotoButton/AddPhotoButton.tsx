import React from 'react';
import {Pressable} from 'react-native';
import Icon from 'react-native-vector-icons/AntDesign';
import styles from './AddPhotoButton.styles';

export type Props = {
  onPress: () => void;
  testID?: string;
  testOnly_pressed?: boolean;
};

export const AddPhotoButton: React.FC<Props> = ({
  onPress,
  testID,
  testOnly_pressed,
}) => {
  return (
    <Pressable
      testID={testID}
      style={({pressed}) => (pressed ? styles.pressed : styles.pressable)}
      onPress={onPress}
      testOnly_pressed={testOnly_pressed}>
      <Icon name={'pluscircle'} style={styles.buttonIcon} />
    </Pressable>
  );
};
