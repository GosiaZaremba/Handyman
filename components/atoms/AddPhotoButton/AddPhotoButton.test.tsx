import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import {AddPhotoButton} from './AddPhotoButton';

const mockOnPress = jest.fn();

describe('Atom AddPhotoButton', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <AddPhotoButton
          onPress={mockOnPress}
          testID="atom-addPhotoButton"
          testOnly_pressed={true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure onPress works correctly', () => {
    const {getByTestId} = render(
      <AddPhotoButton
        onPress={mockOnPress}
        testID="atom-addPhotoButton"
        testOnly_pressed={false}
      />,
    );
    const addPhotoButton = getByTestId('atom-addPhotoButton');
    fireEvent.press(addPhotoButton);
    expect(mockOnPress).toHaveBeenCalledTimes(1);
    expect(addPhotoButton.props.style.borderRadius).toBe(25);
    expect(addPhotoButton.props.style.opacity).toBe(undefined);
  });

  test('ensure pressable pressed works correctly', () => {
    const {getByTestId} = render(
      <AddPhotoButton
        onPress={mockOnPress}
        testID="atom-addPhotoButton"
        testOnly_pressed={true}
      />,
    );
    const addPhotoButton = getByTestId('atom-addPhotoButton');
    fireEvent.press(addPhotoButton);
    expect(addPhotoButton.props.style.borderRadius).toBe(25);
    expect(addPhotoButton.props.style.opacity).toBe(0.9);
  });
});
