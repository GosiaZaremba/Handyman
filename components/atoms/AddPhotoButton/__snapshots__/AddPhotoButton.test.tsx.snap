// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom AddPhotoButton Component Snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 25,
      "height": 30,
      "justifyContent": "center",
      "left": 70,
      "opacity": 0.9,
      "position": "relative",
      "top": -30,
      "width": 30,
    }
  }
  testID="atom-addPhotoButton"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#43b35d",
          "fontSize": 26,
        },
        Object {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î˜¦
  </Text>
</View>
`;
